include/simple-mpc/interpolator.hpp

```diff
@@ -0,0 +1,53 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2025, INRIA
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+/**
+ * @file interpolator.hpp
+ * @brief Interpolation class for practical control of the robot
+ */
+
+#ifndef SIMPLE_MPC_INTERPOLATOR_HPP_
+#define SIMPLE_MPC_INTERPOLATOR_HPP_
+
+#include "simple-mpc/fwd.hpp"
+
+namespace simple_mpc
+{
+  class Interpolator
+  {
+  public:
+    explicit Interpolator(const long nx, const long nv, const long nu, const long nf, const double MPC_timestep);
```

Comment:
Depending on what "lifetime" we want to give to the interpolator, the constructor could take as argument only the robot model.

It can also take the time step, but this is debatable. I would rather be in favor of giving the "timestep" when calling `interpollate`